infra/modules/azure_event_hub/main.tf

```diff
@@ -0,0 +1,86 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.111.0"
```

Comment:
Higher marjor version will be incompatible anyway, I'd then add < 4.0.0:

`version = ">= 3.0.0, < 4.0.0"`

Reply:
I don't think it changes much since 3.116 is the last release before 4, and they shouldn't make any more ([versions](https://github.com/hashicorp/terraform-provider-azurerm/releases))